client/my-sites/themes/install-theme-button.tsx

```diff
@@ -2,17 +2,33 @@ import { FEATURE_INSTALL_THEMES } from '@automattic/calypso-products';
 import { Button } from '@automattic/components';
 import { translate } from 'i18n-calypso';
 import { connect } from 'react-redux';
+import { Dispatch } from 'redux';
 import { recordTracksEvent } from 'calypso/state/analytics/actions';
 import { isUserLoggedIn } from 'calypso/state/current-user/selectors';
 import isAtomicSite from 'calypso/state/selectors/is-site-automated-transfer';
 import siteHasFeature from 'calypso/state/selectors/site-has-feature';
 import { isJetpackSiteMultiSite, isJetpackSite } from 'calypso/state/sites/selectors';
+import { IAppState } from 'calypso/state/types';
 import { getSelectedSiteId, getSelectedSiteSlug } from 'calypso/state/ui/selectors';
 import { trackClick } from './helpers';
-
 import './install-theme-button.scss';
 
-const InstallThemeButton = ( {
+interface TracksEventProps {
+	tracksEventProps: { site_type: string | null };
+}
+
+interface InstallThemeButtonProps {
+	isMultisite: boolean | null;
+	jetpackSite: boolean | null | undefined;
+	isLoggedIn: boolean;
+	siteSlug: string | null;
+	siteCanInstallThemes: boolean | null;
+	dispatchTracksEvent( { tracksEventProps }: TracksEventProps ): void;
+	canUploadThemesOrPlugins: boolean;
+	atomicSite: boolean | null;
+}
```

Comment:
A better way to type components that receive props from `react-redux` `connect` is to use this approach:

```ts
const connector = connect( mapStateToProps, mapDispatchToProps );

type PropsFromRedux = ConnectedProps< typeof connector >;

export default connector( InstallThemeButton );
```

Here, `PropsFromRedux` infers props automatically, you don't need to redefine them.

Then at the top, you can use this

```ts
interface InstallThemeButtonProps extends PropsFromRedux {
	canUploadThemesOrPlugins?: boolean;
}
```

Reply:
🤯 🤯 🤯

Reply:
So cool! Done in [33618a8](https://github.com/Automattic/wp-calypso/pull/61054/commits/33618a84b197a47cc5c000041ea2d73549bf3dc1)